### DISCUSSION TITLE
Failed to execute goal org.jreleaser:jreleaser-maven-plugin:1.4.0:deploy (default-cli) on project relewise-sdk: Execution default-cli of goal org.jreleaser:jreleaser-maven-plugin:1.4.0:deploy failed: JReleaser has not been properly configured.

### DISCUSSION THREAD

**Discussion by KSGRelewise** [jreleaser/jreleaser#1216] [Q&A] [answered]
Hey,

I'm getting the following error when trying to run my CI/CD script which uses the JReleaser Maven plugin:
```bash
Error:  Failed to execute goal org.jreleaser:jreleaser-maven-plugin:1.4.0:deploy (default-cli) on project relewise-sdk: Execution default-cli of goal org.jreleaser:jreleaser-maven-plugin:1.4.0:deploy failed: JReleaser has not been properly configured.
Error:  java.lang.NullPointerException: Cannot invoke "org.jreleaser.model.internal.release.BaseReleaser.getRepoUrl()" because the return value of "org.jreleaser.model.internal.release.Release.getReleaser()" is null
Error:  
Error:  -> [Help 1]
```
The specific CI/CD run that gave the above error can be seen here: https://github.com/Relewise/relewise-sdk-java/actions/runs/4242401763/jobs/7373852776
I'm a bit of a novice in releasing Java Maven packages, so I'm sure it is something I'm doing wrong or some configuration that I'm missing.
Any help would be much appreciated. 😊

-Kristoffer

**Comment by aalmiray** [comment, marked as answer]
I've cloned the project and found 3 things that must be updated:

1. Staged artifacts must be placed inside `src/target/staging-repository` otherwise the pom file won't find them
```
mvn -Prelease deploy jreleaser:deploy -DaltDeploymentRepository=local::file:src/target/staging-deploy --file src/pom.xml
```
2. The pom file is not located at the root of the git repository which causes the NPE. this can be fixed by adding a `<gitRootSearch>` property to the jreleaser's plugin configuration, such as
```xml
<plugin>
    <groupId>org.jreleaser</groupId>
    <artifactId>jreleaser-maven-plugin</artifactId>
    <version>1.4.0</version>
    <configuration>
        <gitRootSearch>true</gitRootSearch>
        <jreleaser>
        ...
```
3. Lastly, JReleaser will complain that the project is not fully configured
```
[ERROR] == JReleaser ==
[ERROR] project.copyright must not be blank
```
This can be solved by adding `<inceptionYear>` to the main pom elements, such as
```xml
    <name>relewise-sdk</name>
    <inceptionYear>2023</inceptionYear>
    <url>https://github.com/Relewise/relewise-sdk-java</url>
```

**Reply by KSGRelewise** [reply]
A million times, thanks! Love from here. Keep up the good work with this great tool.
